### PR TITLE
Fix bug in eliminate_contiguous for operators that have module inputs

### PR DESCRIPTION
**Pull request by pfultz2**

### CODE REVIEW

File: test/eliminate_contiguous_test.cpp [hunk 1]
```diff
@@ -196,15 +196,47 @@ TEST_CASE(contiguous_pointwise)
...
-    EXPECT(std::distance(mm->begin(), mm->end()) == (count - 1));
+    EXPECT(std::distance(mm->begin(), mm->end()) == (count - 2));
     EXPECT(std::none_of(
         mm->begin(), mm->end(), [](auto&& ins) { return ins.name() == "contiguous"; }));
 }
 
+TEST_CASE(contiguous_nhwc_pointwise)
+{
+    auto s = migraphx::shape::from_permutation(
+        migraphx::version_1::shape::float_type, {2, 3, 8, 8}, {0, 2, 3, 1});
```
Comment (umangyadav) [inline test/eliminate_contiguous_test.cpp:212]: Looks like vscode auto-gen. Fix tidy.

**Review by umangyadav — APPROVED** [review]
Surprised, this didn't show up earlier as a bug so far.

File: test/eliminate_contiguous_test.cpp (hunk 1 above)
Comment (pfultz2) [reply test/eliminate_contiguous_test.cpp:212]: Oops. Actually, I use sublime not vscode. Either way its from the clangd autocompletions.

**Review by kahmed10 — APPROVED** [review]

**Comment by causten** [comment]
@pfultz2 onnx-taau-downsample dropped significantly, can you review

**Comment by pfultz2** [comment]
> onnx-taau-downsample dropped significantly, can you review

So this doesn't seem to run with miopen on my local machine. After using #2102(which fixes this issue), the compiled programs is the same as the program on develop. Most likely a mixed layout is being used and miopen is calling a different kernel, which most likely produces incorrect results.

**Pull request merged**